feat(app): Show and convert fiats if setting is enabled

lisabaut opened the pull request:
### Describe the changes you have made in this PR

This is the main integration of the new settings feature.

This PR does:
- add a new boolean to the DEFAULT_SETTINGs => `showFiat`
- `showFiat` is `true` by default
- ~Settings: If in development mode show a "Show Fiat" toggle~ => Toggle is shown in production
- Settings: If `showFiat` is `false` => hide settings for "currency"  and "exchange" 
- Add new SettingsContext and Provider which gets the current settings from `api.getSettings` 
- Wrap all screens with this Provider
- Settings: get settings now via SettingsContext and not directly from `api.getSettings` no more
- Settings: ensure to update the SettingsContext state too when saving new settings
- Acounts: Divide `getSats` and `getFiatValues`
- Acounts: get `showFiat`  from SettingsContext
- Accounts: do not fetch fiats if `showFiat` is `false` => fiatValue remains or becomes an empty string
- Do the same for all other places where `getFiatValue` is used (Receive, Pay, Payment, KeySend, Home, Publisher)
- Ensure to not render any data if `fiatValues` is an empty string (AccountMenu, DualCurrency, Transactions)
- addes settings fixtures for tests
- and some minor refactorings

Please checkout this branch and toggle the Setting by yourself to check if something is missing 🙏 


### Next steps:
- [x] talk with Design about the Toggle in Settings view => waiting for update in ticket
- [x] show the Toggle in production mode => **FEEDBACK REQUIRED** if this PR should already render the toggle for production users
- [x] more usages of the hook `useSettings` on component level, see [comment](https://github.com/getAlby/lightning-browser-extension/pull/1177#discussion_r929754374) => new ticket https://github.com/getAlby/lightning-browser-extension/issues/1210

### Link this PR to an issue

Part of https://github.com/getAlby/lightning-browser-extension/issues/1141

### How has this been tested?

I added addiitonal tests where a Test File existed already.

| Setting enabled  | Setting disabled  |
|---|---|
|<img src="https://user-images.githubusercontent.com/11243967/180760429-09f03585-d5c3-42b3-bd80-d0b57b589ddd.png" width="300" />   |  <img src="https://user-images.githubusercontent.com/11243967/180760432-3da38487-7af9-4e0f-b31e-6c6ec3cea992.png" width="300" />  |
|<img src="https://user-images.githubusercontent.com/11243967/180765956-a94f31d5-6328-427c-849b-fdc0e523c124.png" width="300" />    |  <img src="https://user-images.githubusercontent.com/11243967/180765961-5ec97819-3b2f-4535-a5b6-c1c1343b6be4.png" width="300" />     |
|<img src="https://user-images.githubusercontent.com/11243967/180768822-59767884-7bca-4449-ba5d-8233cea5e5c1.png" width="300" />    |  <img src="https://user-images.githubusercontent.com/11243967/180768828-9f0f50e8-7216-4b1e-8bb3-d0faeb1a35d3.png" width="300" />     |
|<img width="370" alt="incoming_with" src="https://user-images.githubusercontent.com/11243967/181722611-82ceb402-9323-4ebb-9872-8afd00cdcca5.png">  |  <img width="369" alt="incoming_without" src="https://user-images.githubusercontent.com/11243967/181722616-5960338c-c0ac-4ea1-9f22-b90f727a7123.png">  |





### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [ ] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

escapedcat commented on src/app/components/AccountMenu/index.tsx:
```diff
@@ -68,14 +68,16 @@ function AccountMenu({ title, balances, showOptions = true }: Props) {
-            <span className="text-xs text-gray-600 dark:text-neutral-400">
-              ~{balances.fiatBalance}
-            </span>
+            {!!balances.fiatBalance && (
```
Wondering if when this is active and the API takes a bit longer, will the loading-skeleton is already gone but the fiat value might still be i.e. `~0$`?

lisabaut replied:
Can you explain again what this Skeleton is good for? So far I did not experience any loading issues.

escapedcat replied:
Not blocking. We can have a look once this is merged.

escapedcat reviewed (commented):
Really awesome PR! Thanks!

Checked it out. Not sure if everything works as supposed or if I'm doing something wrong:
![image](https://user-images.githubusercontent.com/1016218/180119519-05b5497c-5d88-4117-a605-f6f86ded899c.png)
After switching the toggle the AccountMenu still shows Fiat. Also after a hard reload. Does this work for you? 

Same for the Popup:
![image](https://user-images.githubusercontent.com/1016218/180119990-55ff828d-48e6-4939-89a7-f2153ccd5206.png)

Weird

escapedcat commented on src/app/components/AllowanceMenu/index.test.tsx:
```diff
@@ -1,10 +1,22 @@
 
+const mockSettings = {
+  showFiat: true,
+} as SettingsStorage;
```
Would be nice if we could avoid `as`. Would it work to define all needed props and do `mockSettings:SettingsStorage = {`? Same for the other usages.

lisabaut replied:
Changes everywhere with adding a new fixtures folder for tests (=> see also the discussion [here](https://github.com/getAlby/lightning-browser-extension/pull/968#discussion_r928803305))

escapedcat commented on src/app/components/AllowanceMenu/index.tsx:
```diff
@@ -2,6 +2,7 @@ import { GearIcon } from "@bitcoin-design/bitcoin-icons-react/filled";
 import { CrossIcon } from "@bitcoin-design/bitcoin-icons-react/outline";
 import { useState, useEffect } from "react";
 import Modal from "react-modal";
+import { useSettings } from "~/app/context/SettingsContext";
```
Maybe we could also replace this then? Not in this but in a next PR?:
https://github.com/getAlby/lightning-browser-extension/blob/feat/738_openchannel/src/app/screens/LNURLPay.tsx/#L100-L109

lisabaut replied:
Yes! Is noted and added to the TODOs above in the description

bumi replied:
+ the other occurrences where we use `api.getSettings` (currencyConvert.ts, LNURLPay.tsx, utils/index.ts, batteries/index.ts, ...(?)

lisabaut replied:
The new hook `useSettings` works only on component /rendering level

I created a new ticket here as a follow up
https://github.com/getAlby/lightning-browser-extension/issues/1210

bumi commented on src/app/components/form/DualCurrencyField/index.tsx:
```diff
@@ -4,7 +4,7 @@ import { classNames } from "~/app/utils";
   suffix?: string;
   endAdornment?: React.ReactNode;
-  fiatValue: string | number;
+  fiatValue: string;
```
that means we will always have a formatted string in there? 

handling amounts is something I think we need to be careful to not be confusing: 
where do we use strings, numbers, bignumbers, ...? 
is it formatted? how to we calculate with it?

escapedcat replied:
This is a valid correction I believe. They value is initaly being created by `getFiatValue` which always returns a string anyway.

lisabaut replied:
Yes exactly. Every fiatValue is [formatted here](https://github.com/getAlby/lightning-browser-extension/blob/665ccd0cf611e31f82b07a9a2953707d22bd9db9/src/common/utils/currencyConvert.ts#L96-L111) which always returns a string.

bumi commented on src/app/screens/Settings.tsx:
```diff
@@ -220,12 +153,83 @@ function Settings() {
           )}
         </Setting>
+
+        {process.env.NODE_ENV === "development" && (
```
+1 show it in production mode :shipit:

lisabaut replied:
Done ✅

bumi reviewed (approved):
enable it in production mode and then we merge this :shipit: 

what are the downsides of using the context? 
normally when the context changes everything will be re-rendered, correct? 
in the settings context the settings data also does not often change so this is not a problem? - in fact it actually only changes from the settings page currently when `updateSetting` is called? 

can you take care of changing the other usages of the settings that we there also use this new context?

bumi commented:
can the langauge also be changed from that settings context? 
https://github.com/getAlby/lightning-browser-extension/pull/869

so whenever the settings are loaded we call that `changeLanguage` function

escapedcat commented:
> can the language also be changed from that settings context? #869
> 
> so whenever the settings are loaded we call that `changeLanguage` function

Yeah, we can take a look once this is merged

lisabaut commented:
> enable it in production mode and then we merge this :shipit:
> 

=> done

> what are the downsides of using the context? normally when the context changes everything will be re-rendered, correct? in the settings context the settings data also does not often change so this is not a problem? - in fact it actually only changes from the settings page currently when `updateSetting` is called?
> 

=> yes, all components which read from the settings context with the hook `useSettings` will re-rerender. Which is is intended in this case

> can you take care of changing the other usages of the settings that we there also use this new context?

=> sure, to stay in scope I would do this in follow-up PRs . I assigned myself to this [ticket](https://github.com/getAlby/lightning-browser-extension/issues/1210), feel free to create more tickets and assign me if you have more places in mind where this context should be used.

lisabaut commented:
I rebased again and fixed the new view for Incoming Transactions too. 
If the setting "sho fiat" is disabled => no fiats in incoming transactions.
See two new screenshots above.
